crates/evm/src/evm/handler.rs

```diff
@@ -401,8 +405,12 @@ impl<SPEC: Spec, EXT: CitreaExternalExt, DB: Database> CitreaHandler<SPEC, EXT,
         context: &mut Context<EXT, DB>,
         result: FrameResult,
     ) -> Result<ResultAndState, EVMError<<DB as Database>::Error>> {
-        let diff_size =
-            calc_diff_size::<EXT, SPEC, DB>(context).map_err(EVMError::Database)? as u64;
+        let uncompressed_size =
+            calc_diff_size::<EXT, SPEC, DB>(context).map_err(EVMError::Database)?;
```

Comment:
you should check if Spec::Cancun is enabled here. 

On Fork1, we'll set cancun to enabled in revm, so internally actions can be branched on that.


why? imagine you're syncing to testnet with this code. you'll apply different balance changes to all txs, causing you to be on a completely different state than all other nodes

Reply:
made changes